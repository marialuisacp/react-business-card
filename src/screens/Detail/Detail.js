import React, { Component } from 'react';
import { withRouter } from "react-router-dom";

import '../../assets/styles/styles.scss';
import './Detail.scss';

class Detail extends Component {
  goToBack = () => {
    this.props.history.push('/');
  }

  componentWillMount() {
    const hasParam = (this.props.location.state && this.props.location.state.data);

    if (!hasParam) {
      this.goToBack();
    }
  };

  render() {
    const person = (this.props.location.state && this.props.location.state.data)
      ? this.props.location.state.data
      : {
        name: '',
        email: '',
        office: '',
        imaeg: ''
      };

    return (
      <div className='screen' id='detail-screen'>
        <div className='content'>
          <div className='btn-back' onClick={this.goToBack}></div>
          <div className='info'>
            <div className='avatar'>
              <img alt='Foto do profissional' src={person.image} />
            </div>
            <p className='title'>
              {person.name}
            </p>
            <p>
              {person.office}
            </p>
          </div>
          <div className='card-company'>
            <div className='info-superior'>
              <div className='logo'>
                <img alt='logo da empresa' src='./assets/images/logo.png' />
              </div>
              <div className='phones'>
                <p> 31 | 9.9147-8701 </p>
                <p> 31 | 9.9147-2020 </p>
              </div>
            </div>
            <div className='info-inferior'>
              <p> {person.email} </p>
              <div className='line'></div>
            </div>
          </div>
          <div className='card-qrcode'>
            <img alt='qrcode' src='./assets/images/qrcode.png' />
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(Detail);