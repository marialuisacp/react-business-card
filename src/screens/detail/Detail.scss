$avatar-width: 160px;
#detail-screen {
  background-image: url(/images/bg_vertical.jpg);
  background-position: center;
  background-size: auto 100%;

  .btn-back {
    background-image: url(/images/arrow_left.png);
    background-position: center;
    background-size: auto 50%;
    background-repeat: no-repeat;
    width: 60px;
    height: 60px;
    cursor: pointer;
    position: absolute;
  } 

  .info {
    text-align: center;
    position: relative;
    top: 50px;

    .avatar {
      display: block;
      border-radius: 50%;
      width: $avatar-width;
      height: $avatar-width;
      overflow: hidden;
      margin: 0 auto;

      img {
        max-width: 100%;
      }
    }
    p {
      font-size: 20px;
      color: #fff;
      line-height: .5em;
      font-weight: lighter;
    }
    .title {
      font-weight: bold;
    }
  }

  .card-company {
    background-color: #335264;
    border-radius: 24px;
    height: 296px;
    position: relative;
    top: 52px;
    margin: 0 auto;
    max-width: 548px;
    .info-superior {
      display: flex;
      padding-top: 60px;
      .logo {
        height: 100px;
        display: inline-block;
        width: 50%;
        margin: 0;
        text-align: center;
        img {
          display: block;
          margin: 0 auto;
          max-width: 60%;
        }
      }
      .phones {
        display: inline-block;
        height: 100px;
        width: 50%;
        margin: 0;
        text-align: center;
        border-left: solid 2px #fff;
        p {
          color: #fff;
        }
      }
    }

    .info-inferior {
      text-align: center;
      color: #fff;
      width: 94%;
      height: 80px;
      position: relative;
      margin: 0 auto;
      p {
        margin-top: 40px;
        font-size: 18px;
      }
      &:after {
        background: linear-gradient(to right, #66cd2f 60%,#00a9e3 60%, #00a9e3 80%, #fff 80%);
        position: absolute;
        content: '';
        height: 6px;
        right: 0px;
        left: 0px;
        top: 54px;
      }
    }
  }

  .card-qrcode {
    background: #fff;
    width: 296px;
    height: 296px;
    overflow: hidden;
    margin: 0 auto;
    margin-top: 70px;
    border-radius: 20px;
    margin-bottom: 64px;

    img {
      max-width: 84%;
      margin: 0 auto;
      display: block;
      margin-top: 8%;
    }
  }
}

@media (max-width: 664px) {
  #detail-screen {
    .info {
      top: 30px;
      .avatar {
        width: 120px;
        height: 120px;
      }
    }
    .card-company {
      margin: 0px 24px;
      height: 245px;
      position: relative;
      top: 32px;
      .info-superior {
        padding-top: 40px;
      
        .logo, .phones {
          height: 70px;
        }
      }
    }

    .card-qrcode {
      width: 180px;
      height: 180px;
    }
  }
}