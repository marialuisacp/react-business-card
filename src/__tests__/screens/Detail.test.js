import React from 'react';
import renderer from 'react-test-renderer';

import { createLocation } from 'history';
import { MemoryRouter as Router } from 'react-router-dom';
// import { match as routerMatch } from 'react-router';

import Detail from '../../screens/Detail/Detail';

describe('Detail', () => {
  test('Screen Detail should render correctly', () => {
    const location = {
      state: {
        data: {
          letter: 'D',
          people: [{
            name: 'Danilo Rocha',
            office: 'Desenvolvedor',
            image: './assets/images/people/d01.jpg',
            email: 'danilo.rocha@mundiale.com'
          }]
        }
      }
    };

    const component = renderer.create(
      <Router>
        <Detail location={createLocation(location)} />
      </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});
