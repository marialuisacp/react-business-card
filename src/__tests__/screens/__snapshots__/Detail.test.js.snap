// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Screen Detail should render correctly 1`] = `
<div
  className="screen"
  id="detail-screen"
>
  <div
    className="content"
  >
    <div
      className="btn-back"
      onClick={[Function]}
    />
    <div
      className="info"
    >
      <div
        className="avatar"
      >
        <img
          alt="Foto do profissional"
        />
      </div>
      <p
        className="title"
      >
        
      </p>
      <p>
        
      </p>
    </div>
    <div
      className="card-company"
    >
      <div
        className="info-superior"
      >
        <div
          className="logo"
        >
          <img
            alt="logo da empresa"
            src="./assets/images/logo.png"
          />
        </div>
        <div
          className="phones"
        >
          <p>
             31 | 9.9147-8701 
          </p>
          <p>
             31 | 9.9147-2020 
          </p>
        </div>
      </div>
      <div
        className="info-inferior"
      >
        <p>
           
          
           
        </p>
        <div
          className="line"
        />
      </div>
    </div>
    <div
      className="card-qrcode"
    >
      <img
        alt="qrcode"
        src="./assets/images/qrcode.png"
      />
    </div>
  </div>
</div>
`;
